lightning/src/ln/channelmanager.rs

```diff
@@ -2159,7 +2165,7 @@ where
 			payment_hash,
 			incoming_shared_secret: shared_secret,
 			incoming_amt_msat: Some(amt_msat),
-			outgoing_amt_msat: amt_msat,
+			outgoing_amt_msat: hop_data.amt_to_forward,
```

Comment:
nit: I think it would be helpful to document on `PendingHTLCInfo` that on receive, `incoming_amt_msat` refers to the actual amount received and `outgoing_amt_msat` refers to the sender-intended amount